framework/db/Command.php

```diff
@@ -829,30 +829,32 @@ public function execute()
     {
         $sql = $this->getSql();
 
-        $rawSql = $this->getRawSql();
-
-        Yii::info($rawSql, __METHOD__);
+        $log = $this->db->enableQueryLog;
+        if ($log) {
+            $rawSql = $this->getRawSql();
+            Yii::info($rawSql, __METHOD__);
+            $token = $rawSql;
+        }
 
         if ($sql == '') {
             return 0;
         }
 
         $this->prepare(false);
 
-        $token = $rawSql;
         try {
-            Yii::beginProfile($token, __METHOD__);
+            $log && Yii::beginProfile($token, __METHOD__);
```

Comment:
If we consider this as convention, it does not matter if code works diferentlly or not. Same as 

```php
if (true) {
    $this->doSomething();
}
```
works the same as 
```php
if (true)
    $this->doSomething();
```

but only first version is preferred.